yarn-project/types/src/logs/log_id.ts

```diff
@@ -0,0 +1,87 @@
+import { toBufferBE } from '@aztec/foundation/bigint-buffer';
+import { BufferReader } from '@aztec/foundation/serialize';
+
+/** A globally unique log id. */
+export class LogId {
+  /**
+   * Parses a log id from a string.
+   * @param blockNumber - The block number.
+   * @param txIndex - The transaction index.
+   * @param logIndex - The log index.
+   */
+  constructor(
+    /** The block number the log was emitted in. */
+    public readonly blockNumber: number,
+    /** The index of a tx in a block the log was emitted in. */
+    public readonly txIndex: number,
+    /** The index of a log the tx was emitted in. */
+    public readonly logIndex: number,
+  ) {
+    if (!Number.isInteger(blockNumber)) {
+      throw new Error(`Invalid block number: ${blockNumber}`);
+    }
+    if (!Number.isInteger(txIndex)) {
+      throw new Error(`Invalid tx index: ${txIndex}`);
+    }
+    if (!Number.isInteger(logIndex)) {
+      throw new Error(`Invalid log index: ${logIndex}`);
+    }
```

Comment:
I believe we're not validating input args anywhere else in `types`. Curious: why on this one?

Reply:
It's on the user input in the CLI [here](https://github.com/AztecProtocol/aztec-packages/blob/d63e6d57982c2d7aa34480cac542fe275ffb0df4/yarn-project/cli/src/utils.ts#L238) and it makes sense to do the checks in the constructor.